src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -112,28 +168,43 @@ abstract class LazyModule()(implicit val p: Parameters)
     children.filter(!_.omitGraphML).foreach { c => c.edgesGraphML(buf, pad) }
   }
 
+  /** execute function to [[children]].
+    * @param iterfunc function to run
+    * */
   def nodeIterator(iterfunc: (LazyModule) => Unit): Unit = {
     iterfunc(this)
     children.foreach( _.nodeIterator(iterfunc) )
   }
 
+  /** get [[children]] */
   def getChildren = children
 }
 
 object LazyModule
 {
+  /** current [[LazyModule]] scope, specifically, it is a stack of [[LazyModule]].
+    * it will be dynamically set by [[LazyScope.apply]] and [[LazyModule.apply]],
+    * */
   protected[diplomacy] var scope: Option[LazyModule] = None
+  /** index of [[LazyModule]], notice there is no No.0 module. */
   private var index = 0
 
+  /** add a [[LazyModule]]
```

Comment:
add to what?

Reply:
changed to 
```
  /** any instance of [[LazyModule]] should be wrapper in to a [[LazyModule.apply]],
    * making use this singleton manage scope and index for elaboration.
    * @param bc [[LazyModule]] instance needed to be wrapped.
    * @param valName [[ValName]] used to naming this val,
    *                it can be automatically generated by [[ValName]] macro, or specify manually.
    * @param sourceInfo [[SourceInfo]] to generate this [[LazyModule]]
    * */
```